src/tests/encore/parametric/methodPolyClosure.enc

```diff
@@ -0,0 +1,18 @@
+read class Bar[t]
+  def bar() : unit
+    println("In Bar")
+  end
+end
+
+read class Foo
+  def foo[ty]() : () -> unit
+    fun() => (new Bar[ty]()).bar()
```

Comment:
Yes, I have not had time to fix it. The discussion below is all there is so far.